instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/InstrumentationType.java

```diff
@@ -0,0 +1,31 @@
+package io.opentelemetry.instrumentation.api.tracer;
+
+public class InstrumentationType {
```

Comment:
It seems to me that `InstrumentationType` is a more generic version of `ServerSpan`/`ClientSpan` - perhaps we should get rid of those? Instead of having two mechanisms that work in a very similar way we'd have just one.